### PR TITLE
Generate presubmit coverage report using CodeCov

### PR DESCRIPTION
**Pull request by tcnghia**
# Description

Generate presubmit coverage report using CodeCov GitHub action.

## Issue reference

Please reference the issue this PR will close: #2069 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: .github/workflows/dapr.yml [hunk 1]
```diff
@@ -60,8 +60,13 @@ jobs:
         if: matrix.target_arch != 'arm' && matrix.target_os != 'windows'
         run: make lint
       - name: Run make test
+        env:	
+          COVERAGE_OPTS: "-coverprofile=coverage.txt -covermode=atomic"
         if: matrix.target_arch != 'arm'
         run: make test
+      - name: Codecov
+        if: matrix.target_arch != 'arm'
```
Comment (youngbupark) [inline .github/workflows/dapr.yml:68]: I think we need code coverage only one time. e.g. linux amd64
so in this case, I would set `if` condition to `matrix.target_arch == 'amd64' && matrix.target_os == 'linux'`

File: .codecov.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ignore:
+  - "**/zz_generated*.go" # Ignore generated files.
```
Comment (youngbupark) [inline .codecov.yaml:2]: Please exclude the below:
`**/pkg/apis/**` : CRD related files including generated ones. 
`**/pkg/proto/**` : grpc protobuf client for dapr
`**/pkg/testing/**` : testing mock

**Review by youngbupark — CHANGES_REQUESTED** [review]
Thanks for this work! this is great to see the test coverage of dapr/dapr.

cc/ @yaron2 @mchmarny @msfussell @orizohar

**Comment by mchmarny** [comment]
Add badge to readme 

```rm
[![codecov](https://codecov.io/gh/dapr/dapr/branch/master/graph/badge.svg)](https://codecov.io/gh/dapr/dapr)
```

File: .codecov.yaml (hunk 2 above)
Comment (tcnghia) [reply .codecov.yaml:2]: Done

File: .github/workflows/dapr.yml (hunk 1 above)
Comment (tcnghia) [reply .github/workflows/dapr.yml]: Done

File: .codecov.yaml (hunk 2 above)
Comment (tcnghia) [reply .codecov.yaml:2]: Done

**Review by youngbupark — APPROVED** [review]

**Comment by tcnghia** [comment]
> Add badge to readme
> 
> ```
> [codecov](https://codecov.io/gh/dapr/dapr)
> ```

@mchmarny Done! Thanks

**Review by mchmarny — APPROVED** [review]
/lgtm

**Comment by tcnghia** [comment]
@youngbupark I just made some change in the Markdown file (moving Cov badge to the same location with other badges). Can you please take another look? Thanks!

**Review by youngbupark — APPROVED** [review]

**Pull request merged**